# Título

## Antonio Drago Caetano

https://www.linkedin.com/in/antoniodragoc/

https://github.com/dragoc1

# Introdução

# Descrição dos dados

# Objetivos e análises

# Importação das Bibliotecas

In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


In [5]:
import warnings

#ignorar os avisos do python
warnings.filterwarnings('ignore')

# Definindo as funções

# Importação dos Dados

In [7]:
from zipfile import ZipFile
zf = ZipFile(r'./dados/archive.zip')
dados = pd.read_excel(zf.open('Kaggle_Sirio_Libanes_ICU_Prediction.xlsx'))

# Limpeza dos Dados

# Análise exploratória

# Modelos de Machine Learning

# Tunning do melhor modelo

# Referências

https://www.kaggle.com/harriwashere/task-1-eda-3-model-ensemble

https://www.kaggle.com/felipeoliveiraml/stacking-gradient-boosting-pr-80

https://www.kaggle.com/carlolepelaars/quick-eda-exploratory-data-analysis

https://www.kaggle.com/fernandoramacciotti/interpretable-icu-risk-0-2-window-only

https://github.com/HenriqueCCdA/BC_DS_Projeto_Final

## Artigos
https://neptune.ai/blog/f1-score-accuracy-roc-auc-pr-auc

In [156]:
dados['ICU'].value_counts()

0    1410
1     515
Name: ICU, dtype: int64

In [157]:
# Removendo os valores nulos
dados_limpos = dados.dropna()

# Para pegar somente as tabelas que possuem valores numéricos iremos utilizar dados.describe().columns
x = dados_limpos[dados_limpos.describe().columns].drop(['ICU'], axis=1)
y = dados_limpos['ICU']

In [158]:
x.head()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65  GENDER  DISEASE GROUPING 1  \
4                          0            1       0                 0.0   
6                          1            1       1                 0.0   
8                          1            1       1                 0.0   
9                          1            1       1                 0.0   
14                         2            0       0                 0.0   

    DISEASE GROUPING 2  DISEASE GROUPING 3  DISEASE GROUPING 4  \
4                  0.0                 0.0                 0.0   
6                  0.0                 0.0                 0.0   
8                  0.0                 0.0                 0.0   
9                  0.0                 0.0                 0.0   
14                 0.0                 0.0                 0.0   

    DISEASE GROUPING 5  DISEASE GROUPING 6  HTN  ...  HEART_RATE_DIFF  \
4                  1.0                 1.0  0.0  ...         0.114504   
6                  0.0                 0.0  1.0  ...        -1.000000   
8                  0.0                 0.0  1.0  ...        -0.923664   
9                  1.0                 0.0  1.0  ...         0.114504   
14                 0.0                 0.0  0.0  ...        -0.419847   

    RESPIRATORY_RATE_DIFF  TEMPERATURE_DIFF  OXYGEN_SATURATION_DIFF  \
4                0.176471         -0.238095               -0.818182   
6               -1.000000         -1.000000               -1.000000   
8               -0.764706         -0.880952               -1.000000   
9                0.764706          0.142857               -0.797980   
14              -0.705882         -0.500000               -0.898990   

    BLOODPRESSURE_DIASTOLIC_DIFF_REL  BLOODPRESSURE_SISTOLIC_DIFF_REL  \
4                          -0.389967                         0.407558   
6                          -1.000000                        -1.000000   
8                          -0.906832                        -0.831132   
9                           0.315690                         0.200359   
14                         -0.612422                        -0.343258   

    HEART_RATE_DIFF_REL  RESPIRATORY_RATE_DIFF_REL  TEMPERATURE_DIFF_REL  \
4             -0.230462                   0.096774             -0.242282   
6             -1.000000                  -1.000000             -1.000000   
8             -0.940967                  -0.817204             -0.882574   
9             -0.239515                   0.645161              0.139709   
14            -0.576744                  -0.695341             -0.505464   

    OXYGEN_SATURATION_DIFF_REL  
4                    -0.814433  
6                    -1.000000  
8                    -1.000000  
9                    -0.802317  
14                   -0.900129  

[5 rows x 228 columns]

In [159]:
from sklearn.linear_model import LogisticRegression

modelo = LogisticRegression()
modelo.fit(x,y)


LogisticRegression()

In [160]:
x.iloc[4]

PATIENT_VISIT_IDENTIFIER           2.000000
AGE_ABOVE65                        0.000000
GENDER                             0.000000
DISEASE GROUPING 1                 0.000000
DISEASE GROUPING 2                 0.000000
                                     ...   
BLOODPRESSURE_SISTOLIC_DIFF_REL   -0.343258
HEART_RATE_DIFF_REL               -0.576744
RESPIRATORY_RATE_DIFF_REL         -0.695341
TEMPERATURE_DIFF_REL              -0.505464
OXYGEN_SATURATION_DIFF_REL        -0.900129
Name: 14, Length: 228, dtype: float64

In [161]:
# Testando a precisão do modelo. Importante passar o x.iloc dentro do colchetes pois se não vai passar os valores das colunas somente, então o modelo não entenderá o que é para fazer
modelo.predict([x.iloc[4]])

array([1], dtype=int64)

In [162]:
y.iloc[4]

1

Podemos observar que o modelo teve sucesso em prever que de fato o paciente em questão foi para a UTI

In [163]:
#display(print('O modelo acertou em: %d dos casos' % (sum(modelo.predict(x[:20]) == y[:20]) )))
display(print('O modelo acertou em: {} dos casos'.format(sum(modelo.predict(x[:20]) == y[:20]) )))
modelo.predict(x[:20]) == y[:20]

O modelo acertou em: 20 dos casos


None

4     True
6     True
8     True
9     True
14    True
19    True
22    True
24    True
29    True
34    True
36    True
39    True
44    True
49    True
50    True
54    True
59    True
64    True
69    True
70    True
Name: ICU, dtype: bool

In [164]:
#display(print('O modelo acertou em: %d dos casos' % (sum(modelo.predict(x[:20]) == y[:20]) )))
print('O modelo acertou em: {} dos casos e errou em {}'.format(sum(modelo.predict(x) == y),sum(modelo.predict(x) != y) ))
print('A acuracia do modelo foi de {:.2%} '.format(sum(modelo.predict(x) == y)/len(y)))

O modelo acertou em: 497 dos casos e errou em 48
A acuracia do modelo foi de 91.19% 


Separando os dados em treino e teste com a ferramenta train_train_split

In [165]:
from sklearn.model_selection import train_test_split

# Stratificar é manter a proporção com base em alguma coluna, por exemplo, no nosso caso irá manter a proporção de 0 e 1 com base na coluna y
x_train, x_test, y_train, y_test = train_test_split(x,y, stratify = y, random_state = 73246)

In [166]:
import warnings
#ignorar os avisos do python
warnings.filterwarnings('ignore')

from sklearn.metrics import accuracy_score
# Definindo o modelo
modelo = LogisticRegression()

# Treinando o modelo
modelo.fit(x_train, y_train)

# Fazendo as previsões para os dados de teste
yhat = modelo.predict(x_test)

# Comparando o que foi previsto com os valores corretos
print('{:.2%}'.format(accuracy_score(y_test, yhat)))

83.21%


In [167]:
print(y_train.value_counts(normalize = True))
print()
print(y_test.value_counts(normalize = True))

0    0.514706
1    0.485294
Name: ICU, dtype: float64

0    0.518248
1    0.481752
Name: ICU, dtype: float64


In [168]:
dados.head(6)

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65 AGE_PERCENTIL  GENDER  \
0                         0            1          60th       0   
1                         0            1          60th       0   
2                         0            1          60th       0   
3                         0            1          60th       0   
4                         0            1          60th       0   
5                         1            1          90th       1   

   DISEASE GROUPING 1  DISEASE GROUPING 2  DISEASE GROUPING 3  \
0                 0.0                 0.0                 0.0   
1                 0.0                 0.0                 0.0   
2                 0.0                 0.0                 0.0   
3                 0.0                 0.0                 0.0   
4                 0.0                 0.0                 0.0   
5                 0.0                 0.0                 0.0   

   DISEASE GROUPING 4  DISEASE GROUPING 5  DISEASE GROUPING 6  ...  \
0                 0.0                 1.0                 1.0  ...   
1                 0.0                 1.0                 1.0  ...   
2                 0.0                 1.0                 1.0  ...   
3                 0.0                 1.0                 1.0  ...   
4                 0.0                 1.0                 1.0  ...   
5                 0.0                 0.0                 0.0  ...   

   TEMPERATURE_DIFF  OXYGEN_SATURATION_DIFF  BLOODPRESSURE_DIASTOLIC_DIFF_REL  \
0         -1.000000               -1.000000                         -1.000000   
1         -1.000000               -1.000000                         -1.000000   
2               NaN                     NaN                               NaN   
3         -1.000000               -1.000000                               NaN   
4         -0.238095               -0.818182                         -0.389967   
5         -1.000000               -1.000000                         -1.000000   

   BLOODPRESSURE_SISTOLIC_DIFF_REL  HEART_RATE_DIFF_REL  \
0                        -1.000000            -1.000000   
1                        -1.000000            -1.000000   
2                              NaN                  NaN   
3                              NaN                  NaN   
4                         0.407558            -0.230462   
5                        -1.000000            -1.000000   

   RESPIRATORY_RATE_DIFF_REL  TEMPERATURE_DIFF_REL  \
0                  -1.000000             -1.000000   
1                  -1.000000             -1.000000   
2                        NaN                   NaN   
3                        NaN             -1.000000   
4                   0.096774             -0.242282   
5                  -1.000000             -1.000000   

   OXYGEN_SATURATION_DIFF_REL    WINDOW  ICU  
0                   -1.000000       0-2    0  
1                   -1.000000       2-4    0  
2                         NaN       4-6    0  
3                   -1.000000      6-12    0  
4                   -0.814433  ABOVE_12    1  
5                   -1.000000       0-2    1  

[6 rows x 231 columns]

In [169]:
from zipfile import ZipFile
zf = ZipFile(r'./dados/archive.zip')
dados = pd.read_excel(zf.open('Kaggle_Sirio_Libanes_ICU_Prediction.xlsx'))

Podemos observar que a primeira coluna se trata do identificador de visitas, logo não quer dizer que cada linha se trata de um paciente, além do mais, as 5 primeiras linhas apresentam o mesmo valor, logo, é relativo a um paciente que visitou o hospital 5 vezes no período de análise.

In [170]:
def preenche_tabela(dados):
    
    # Lendo a documentação dos dados sabemos que as features contínus são todas da coluna 13 até a anti-penúltima
    features_continuas_colunas = dados.iloc[:,13:-2].columns # No iloc, Não pegar nada do inicio (signifca que estamos pegando todas as linhas) e do final pegar as colunas entre 13 e a -2

    # Agrupando pelo identificador do paciente e preenchendo os valores nulos anteriores(bfill) e posteriores(ffill)
    features_continuas = dados.groupby('PATIENT_VISIT_IDENTIFIER', as_index= False)[features_continuas_colunas].fillna(method = 'bfill').fillna(method = 'ffill')


    # Pegando todas as linhas das colunas 0 até 13
    features_categoricas = dados.iloc[:,:13]
    # Pegando a Saída, todas as linhas das colunas -2 em diantes, ou seja, as duas últimas colunas apenas

    saida = dados.iloc[:,-2:]

    # Juntando as partes dos dados: 1 - parte não alterou nada, 2 - limpou e agrupou os dados e 3 - pegou os dados de saída
    dados_finais = pd.concat([features_categoricas, features_continuas, saida], ignore_index= True, axis  = 1 ) # Como argumento do pd.concat deve ser sempre uma lista, Ignorar o index que não está sendo utilizado (como todos tem o mesmo número de linhas, basta encaixar diretamente) e axis 1 para buscar os dados nas colunas e não nas linhas
    
    dados_finais.columns = dados.columns
    return dados_finais


In [171]:
# Utilizando a função criada para tratar os dados

dados_limpos = preenche_tabela(dados)
dados_limpos.dropna(inplace=True)
dados_limpos.describe()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65       GENDER  DISEASE GROUPING 1  \
count               1920.000000  1920.000000  1920.000000         1920.000000   
mean                 191.981771     0.468750     0.369792            0.108333   
std                  111.312587     0.499152     0.482874            0.310882   
min                    0.000000     0.000000     0.000000            0.000000   
25%                   95.750000     0.000000     0.000000            0.000000   
50%                  191.500000     0.000000     0.000000            0.000000   
75%                  288.250000     1.000000     1.000000            0.000000   
max                  384.000000     1.000000     1.000000            1.000000   

       DISEASE GROUPING 2  DISEASE GROUPING 3  DISEASE GROUPING 4  \
count         1920.000000         1920.000000         1920.000000   
mean             0.028125            0.097917            0.019792   
std              0.165373            0.297279            0.139320   
min              0.000000            0.000000            0.000000   
25%              0.000000            0.000000            0.000000   
50%              0.000000            0.000000            0.000000   
75%              0.000000            0.000000            0.000000   
max              1.000000            1.000000            1.000000   

       DISEASE GROUPING 5  DISEASE GROUPING 6          HTN  ...  \
count         1920.000000         1920.000000  1920.000000  ...   
mean             0.128125            0.046875     0.213021  ...   
std              0.334316            0.211426     0.409549  ...   
min              0.000000            0.000000     0.000000  ...   
25%              0.000000            0.000000     0.000000  ...   
50%              0.000000            0.000000     0.000000  ...   
75%              0.000000            0.000000     0.000000  ...   
max              1.000000            1.000000     1.000000  ...   

       RESPIRATORY_RATE_DIFF  TEMPERATURE_DIFF  OXYGEN_SATURATION_DIFF  \
count            1920.000000       1920.000000             1920.000000   
mean               -0.758150         -0.812388               -0.910214   
std                 0.433588          0.296775                0.255728   
min                -1.000000         -1.000000               -1.000000   
25%                -1.000000         -1.000000               -1.000000   
50%                -1.000000         -1.000000               -1.000000   
75%                -0.705882         -0.690476               -0.919192   
max                 1.000000          1.000000                1.000000   

       BLOODPRESSURE_DIASTOLIC_DIFF_REL  BLOODPRESSURE_SISTOLIC_DIFF_REL  \
count                       1920.000000                      1920.000000   
mean                          -0.818008                        -0.759122   
std                            0.301726                         0.389598   
min                           -1.000000                        -1.000000   
25%                           -1.000000                        -1.000000   
50%                           -1.000000                        -1.000000   
75%                           -0.710805                        -0.622483   
max                            1.000000                         1.000000   

       HEART_RATE_DIFF_REL  RESPIRATORY_RATE_DIFF_REL  TEMPERATURE_DIFF_REL  \
count          1920.000000                1920.000000           1920.000000   
mean             -0.848720                  -0.771187             -0.813295   
std               0.247117                   0.401430              0.295619   
min              -1.000000                  -1.000000             -1.000000   
25%              -1.000000                  -1.000000             -1.000000   
50%              -1.000000                  -1.000000             -1.000000   
75%              -0.764487                  -0.725806             -0.698390   
max               1.000000                   1.000000              

In [172]:
dados_limpos.query("WINDOW == '0-2' and ICU == 1 ")

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65 AGE_PERCENTIL  GENDER  \
5                            1            1          90th       1   
85                          17            0          40th       0   
135                         27            1          60th       1   
205                         41            1          70th       0   
235                         47            0          50th       0   
240                         48            0          60th       0   
270                         54            1          60th       0   
310                         62            1          70th       1   
425                         85            1          70th       0   
450                         90            0          50th       0   
520                        104            1          60th       0   
530                        106            1          70th       0   
545                        109            0          30th       0   
585                        117            1          60th       1   
720                        144            1          80th       0   
880                        176            1          60th       0   
1010                       202            1          60th       0   
1030                       206            0          20th       1   
1130                       226            1    Above 90th       1   
1165                       233            1          90th       0   
1195                       239            0          10th       0   
1305                       261            0          50th       0   
1350                       270            1          70th       0   
1410                       282            1    Above 90th       1   
1430                       286            0          40th       0   
1530                       306            0          10th       0   
1575                       315            0          50th       0   
1645                       329            1          90th       0   
1795                       359            1          80th       0   
1810                       362            0          10th       0   
1830                       366            0          30th       0   
1895                       379            1    Above 90th       1   

      DISEASE GROUPING 1  DISEASE GROUPING 2  DISEASE GROUPING 3  \
5                    0.0                 0.0                 0.0   
85                   0.0                 0.0                 0.0   
135                  0.0                 0.0                 0.0   
205                  0.0                 0.0                 0.0   
235                  0.0                 0.0                 0.0   
240                  0.0                 0.0                 0.0   
270                  0.0                 0.0                 1.0   
310                  0.0                 0.0                 0.0   
425                  0.0                 0.0                 1.0   
450                  0.0                 0.0                 0.0   
520                  0.0                 0.0                 0.0   
530                  0.0                 0.0                 0.0   
545                  0.0                 0.0                 0.0   
585                  0.0                 0.0                 0.0   
720                  0.0                 0.0                 0.0   
880                  0.0                 0.0                 0.0   
1010                 0.0                 0.0                 0.0   
1030                 0.0                 0.0                 0.0   
1130                 0.0                 0.0                 0.0   
1165                 0.0                 0.0                 0.0   
1195                 0.0                 0.0                 0.0   
1305                 0.0                 0.0                 0.0   
1350                 0.0                 0.0                 0.0   
1410                 1.0                 0.0                 0.0   
1430                 0.0    

São 32 pacientes que foram parar na UTI nas duas primeiras horas

In [173]:
# Vamos remover os pacientes que já chegaram no hospital e entraram na UTI
dados_limpos.query("WINDOW == '0-2' and ICU == 1 ")['PATIENT_VISIT_IDENTIFIER'].values # .values no final devolve uma lista com os valores da query

array([  1,  17,  27,  41,  47,  48,  54,  62,  85,  90, 104, 106, 109,
       117, 144, 176, 202, 206, 226, 233, 239, 261, 270, 282, 286, 306,
       315, 329, 359, 362, 366, 379], dtype=int64)

In [174]:
# Vamos remover os pacientes que já chegaram no hospital e entraram na UTI
remover = dados_limpos.query("WINDOW == '0-2' and ICU == 1 ")['PATIENT_VISIT_IDENTIFIER'].values # .values no final devolve uma lista com os valores da query

# Removendo os pacientes que já chegaram e entraram direto para a UTI
dados_limpos = dados_limpos.query("PATIENT_VISIT_IDENTIFIER not in @remover")

In [175]:
def prepare_window(rows):
    if(np.any(rows['ICU'])):
        rows.loc[rows['WINDOW'] == '0-2', 'ICU'] = 1 # Buscando as linhas em que a janela é "0-2" e ICU é 1
    return rows.loc[rows['WINDOW'] == '0-2'] 

In [176]:
dados_limpos = dados_limpos.groupby('PATIENT_VISIT_IDENTIFIER').apply(prepare_window) # Aplicando a função que busca os valores de janela 0-2
dados_limpos.describe()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65      GENDER  DISEASE GROUPING 1  \
count                352.000000   352.000000  352.000000          352.000000   
mean                 192.818182     0.457386    0.380682            0.105114   
std                  110.763735     0.498890    0.486246            0.307136   
min                    0.000000     0.000000    0.000000            0.000000   
25%                   97.750000     0.000000    0.000000            0.000000   
50%                  191.500000     0.000000    0.000000            0.000000   
75%                  289.250000     1.000000    1.000000            0.000000   
max                  384.000000     1.000000    1.000000            1.000000   

       DISEASE GROUPING 2  DISEASE GROUPING 3  DISEASE GROUPING 4  \
count          352.000000          352.000000          352.000000   
mean             0.022727            0.090909            0.019886   
std              0.149245            0.287889            0.139809   
min              0.000000            0.000000            0.000000   
25%              0.000000            0.000000            0.000000   
50%              0.000000            0.000000            0.000000   
75%              0.000000            0.000000            0.000000   
max              1.000000            1.000000            1.000000   

       DISEASE GROUPING 5  DISEASE GROUPING 6         HTN  ...  \
count          352.000000          352.000000  352.000000  ...   
mean             0.127841            0.042614    0.193182  ...   
std              0.334388            0.202272    0.395356  ...   
min              0.000000            0.000000    0.000000  ...   
25%              0.000000            0.000000    0.000000  ...   
50%              0.000000            0.000000    0.000000  ...   
75%              0.000000            0.000000    0.000000  ...   
max              1.000000            1.000000    1.000000  ...   

       RESPIRATORY_RATE_DIFF  TEMPERATURE_DIFF  OXYGEN_SATURATION_DIFF  \
count             352.000000        352.000000              352.000000   
mean               -0.962066         -0.965233               -0.985365   
std                 0.110411          0.086132                0.032640   
min                -1.000000         -1.000000               -1.000000   
25%                -1.000000         -1.000000               -1.000000   
50%                -1.000000         -1.000000               -1.000000   
75%                -1.000000         -1.000000               -0.979798   
max                -0.058824         -0.238095               -0.757576   

       BLOODPRESSURE_DIASTOLIC_DIFF_REL  BLOODPRESSURE_SISTOLIC_DIFF_REL  \
count                        352.000000                       352.000000   
mean                          -0.963091                        -0.958921   
std                            0.106132                         0.117561   
min                           -1.000000                        -1.000000   
25%                           -1.000000                        -1.000000   
50%                           -1.000000                        -1.000000   
75%                           -1.000000                        -1.000000   
max                            0.130435                         0.378657   

       HEART_RATE_DIFF_REL  RESPIRATORY_RATE_DIFF_REL  TEMPERATURE_DIFF_REL  \
count           352.000000                 352.000000            352.000000   
mean             -0.976421                  -0.964390             -0.965725   
std               0.059606                   0.095115              0.084306   
min              -1.000000                  -1.000000             -1.000000   
25%              -1.000000                  -1.000000             -1.000000   
50%              -1.000000                  -1.000000             -1.000000   
75%              -1.000000                  -1.000000             -1.000000   
max              -0.535878                  -0.268817             -0.285468   

     

In [181]:
dados_limpos.head()

PATIENT_VISIT_IDENTIFIER  AGE_ABOVE65  \
PATIENT_VISIT_IDENTIFIER                                             
0                        0                          0            1   
2                        10                         2            0   
3                        15                         3            0   
4                        20                         4            0   
5                        25                         5            0   

                            AGE_PERCENTIL  GENDER  DISEASE GROUPING 1  \
PATIENT_VISIT_IDENTIFIER                                                
0                        0           60th       0                 0.0   
2                        10          10th       0                 0.0   
3                        15          40th       1                 0.0   
4                        20          10th       0                 0.0   
5                        25          10th       0                 0.0   

                             DISEASE GROUPING 2  DISEASE GROUPING 3  \
PATIENT_VISIT_IDENTIFIER                                              
0                        0                  0.0                 0.0   
2                        10                 0.0                 0.0   
3                        15                 0.0                 0.0   
4                        20                 0.0                 0.0   
5                        25                 0.0                 0.0   

                             DISEASE GROUPING 4  DISEASE GROUPING 5  \
PATIENT_VISIT_IDENTIFIER                                              
0                        0                  0.0                 1.0   
2                        10                 0.0                 0.0   
3                        15                 0.0                 0.0   
4                        20                 0.0                 0.0   
5                        25                 0.0                 0.0   

                             DISEASE GROUPING 6  ...  TEMPERATURE_DIFF  \
PATIENT_VISIT_IDENTIFIER                         ...                     
0                        0                  1.0  ...         -1.000000   
2                        10                 0.0  ...         -1.000000   
3                        15                 0.0  ...         -1.000000   
4                        20                 0.0  ...         -0.952381   
5                        25                 0.0  ...         -0.976190   

                             OXYGEN_SATURATION_DIFF  \
PATIENT_VISIT_IDENTIFIER                              
0                        0                -1.000000   
2                        10               -0.959596   
3                        15               -1.000000   
4                        20               -0.979798   
5                        25               -0.979798   

                             BLOODPRESSURE_DIASTOLIC_DIFF_REL  \
PATIENT_VISIT_IDENTIFIER                                        
0                        0                          -1.000000   
2                        10                         -0.515528   
3                        15                         -1.000000   
4                        20                         -1.000000   
5                        25                         -0.860870   

                             BLOODPRESSURE_SISTOLIC_DIFF_REL  \
PATIENT_VISIT_IDENTIFIER                                       
0                        0                         -1.000000   
2                        10                        -0.351328   
3                        15                        -1.000000   
4                        20                        -0.883669   
5                        25                        -0.714460   

                             HEART_RATE_DIFF_REL  RESPIRATORY_RATE_DIFF_REL  \
PATIENT_VISIT_IDENTIFIER                                                      
0                        0             -1.000000              

Agora temos somente 352 duas linhsa com identificador de paciente variando de 0 até 384

In [182]:
# Para pegar somente as tabelas que possuem valores numéricos iremos utilizar dados.describe().columns
x = dados_limpos[dados_limpos.describe().columns].drop(['ICU'], axis=1)
y = dados_limpos['ICU']

In [183]:
y.value_counts()

0    189
1    163
Name: ICU, dtype: int64

In [227]:
from sklearn.model_selection import train_test_split
np.random.seed(748596)
# Stratificar é manter a proporção com base em alguma coluna, por exemplo, no nosso caso irá manter a proporção de 0 e 1 com base na coluna y
x_train, x_test, y_train, y_test = train_test_split(x,y, test_size = 1/3,stratify = y)

In [228]:
import warnings
#ignorar os avisos do python
warnings.filterwarnings('ignore')

from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
from sklearn.linear_model import LogisticRegression
# Definindo o modelo
modelo = LogisticRegression()

# Treinando o modelo
modelo.fit(x_train,y_train)

# Fazendo as previsões para os dados de teste
yhat = modelo.predict(x_test)

# Comparando o que foi previsto com os valores corretos
print('{:.2%}'.format(accuracy_score(y_test, yhat)))

75.42%


In [229]:
print(confusion_matrix(y_test,yhat))

[[51 12]
 [17 38]]


In [230]:
print(classification_report(y_test,yhat))

              precision    recall  f1-score   support

           0       0.75      0.81      0.78        63
           1       0.76      0.69      0.72        55

    accuracy                           0.75       118
   macro avg       0.76      0.75      0.75       118
weighted avg       0.75      0.75      0.75       118

